rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java

```diff
@@ -138,6 +158,201 @@ public static Wallet getFederationsSpendWallet(
+     */
+    private static boolean isAnyUTXOAmountBelowMinimum(
+            ActivationConfig.ForBlock activations,
+            BridgeConstants bridgeConstants,
+            BtcTransaction btcTx,
+            Wallet wallet
+    ){
+        Coin minimumPegInTxValue = getMinimumPegInTxValue(activations, bridgeConstants);
+        return btcTx.getWalletOutputs(wallet).stream().anyMatch(transactionOutput ->
+                transactionOutput.getValue().isLessThan(minimumPegInTxValue)
+        );
+    }
+
+    /**
+     * @param activations
+     * @param bridgeConstants
+     * @param btcTx
+     * @param addresses
+     * @return {@link FastBridgeTxResponseCodes#VALID_TX} if each UTXOs sent to federation isn't less than the minimum,
+     * in case any of UTXO is less than the minimum then it returns
+     * {@link FastBridgeTxResponseCodes#UNPROCESSABLE_TX_UTXO_AMOUNT_SENT_BELOW_MINIMUM_ERROR}.
+     */
+    public static FastBridgeTxResponseCodes validateFastBridgePeginValue(
+        ActivationConfig.ForBlock activations,
+        BridgeConstants bridgeConstants,
+        Context context,
+        BtcTransaction btcTx,
+        List<Address> addresses
+    ) {
+        Coin totalAmount = getAmountSentToAddresses(
+            activations,
+            bridgeConstants.getBtcParams(),
+            context,
+            btcTx,
+            addresses
+        );
+
+        if (totalAmount.equals(Coin.ZERO)) {
+            logger.debug("[validateFastBridgePeginValue] Amount sent can't be 0");
+            return FastBridgeTxResponseCodes.UNPROCESSABLE_TX_VALUE_ZERO_ERROR;
+        }
+
+        if (activations.isActive(RSKIP293)){
+            Coin minimumPegInTxValue = getMinimumPegInTxValue(activations, bridgeConstants);
+            if (totalAmount.isLessThan(minimumPegInTxValue)) {
+                logger.debug("[validateFastBridgePeginValue] Amount sent can't be below the minimum {}",
+                    minimumPegInTxValue.value);
+                return FastBridgeTxResponseCodes.UNPROCESSABLE_TX_AMOUNT_SENT_BELOW_MINIMUM_ERROR;
+            }
+
+            if (isAnyUTXOAmountBelowMinimum(activations, bridgeConstants, context, btcTx, addresses)){
+                logger.debug("[validateFastBridgePeginValue] UTXOs amount sent to federation can't be below the minimum {}.",
+                    minimumPegInTxValue.value);
+                return FastBridgeTxResponseCodes.UNPROCESSABLE_TX_UTXO_AMOUNT_SENT_BELOW_MINIMUM_ERROR;
+            }
+        }
+        return FastBridgeTxResponseCodes.VALID_TX;
+    }
+
+    /**
+     * @param context
+     * @param addresses
+     * @return a simple wallet instance with the give list of address added as watched addresses
+     */
+    private static WatchedBtcWallet createWatchedBtcWalletFromAddresses(Context context, List<Address> addresses) {
+        WatchedBtcWallet wallet = new WatchedBtcWallet(context);
+        long now = Utils.currentTimeMillis() / 1000L;
```

Comment:
Use block.timestamp or a constant if needed. The library doesn't seem to use the value except for logging a warning, but it feels cleaner to avoid System.currentTime

Reply:
If this is only being used for logging a warning, I don't see any issue with continue using it for getting the current milliseconds. In fact, the library does the same internally.
 
 
![image](https://user-images.githubusercontent.com/94651730/165189494-ab3c6c3b-614f-4aeb-a7b7-2c72bdb5a756.png)